platform/ext/target/nordic_nrf/common/native_drivers/spu.h

```diff
@@ -77,4 +77,58 @@ void spu_regions_sram_config_non_secure( uint32_t start_addr, uint32_t limit_add
  */
 void spu_regions_flash_config_non_secure_callable(uint32_t start_addr, uint32_t limit_addr);
 
+/**
+ * \brief Restrict access to peripheral to secure
+ *
+ *  Configure a device peripheral to be accessible from Secure domain only.
+ *
+ * \param periph_base_addr peripheral base address
+ *                         (must correspond to a valid peripheral ID)
+ * \param periph_lock Variable indicating whether to lock peripheral security
+ *
+ * \note
+ * - peripheral shall not be a Non-Secure only peripheral
+ * - DMA transactions are configured as Secure
+ */
+void spu_peripheral_config_secure(uint32_t periph_base_addr, bool periph_lock);
+
+/**
+ * Configure a device peripheral to be accessible from Non-Secure domain.
+ *
+ * \param periph_base_addr peripheral base address
+ *                         (must correspond to a valid peripheral ID)
+ * \param periph_lock Variable indicating whether to lock peripheral security
+ *
+ * \note
+ * - peripheral shall not be a Secure-only peripheral
+ * - DMA transactions are configured as Non-Secure
+ */
+void spu_peripheral_config_non_secure(u32_t periph_base_addr, bool periph_lock);
+
+/**
+ * Configure DPPI channels to be accessible from Non-Secure domain.
+ *
+ * \param dppi_lock Variable indicating whether to lock DPPI channel security
+ *
+ * \note all channels are configured as Non-Secure
+ */
+static inline void spu_dppi_config_non_secure(bool dppi_lock)
```

Comment:
I guess this won't work with armclang, as it seems to use `__inline` instead of `inline`. Use `__STATIC_INLINE`.

Reply:
Well, I'm probably wrong in the above, as we depend on C99 anyway, so there is no difference. Sorry for the noise.

Reply:
Yeap, looks like we do.
And `static inline` is used extensively as is in the code base...